ui/src/main/java/org/pentaho/di/ui/job/entry/JobEntryDialog.java

```diff
@@ -396,21 +416,36 @@ public EditConnectionListener( CCombo wConnection ) {
 
     @Override public void widgetSelected( SelectionEvent e ) {
       DatabaseMeta databaseMeta = jobMeta.findDatabase( wConnection.getText() );
```

Comment:
Sketchy again. Anything that calls jobMeta.findDatabase should be looked at carefully. Collisions only matter at the same level, not in the combined list.

Reply:
I think this call is correct.  It's locating the connection that the user is editing, which should be the one from the flattened list, right?

Reply:
yes, this is correct, good call.